### DISCUSSION TITLE
Cannot login with Readrops client (404 Bad request)

### DISCUSSION THREAD

**Discussion by 360fun** [FreshRSS/FreshRSS#6971] [Q&A] [answered]
Hi, I love FreshRSS but I cannot login with the Readrops client. 😰

I'm waiting the reply from the dev but maybe you also have some clues on how to solve the problem:
https://github.com/readrops/Readrops/issues/98

Thanks 🙏

**Comment by Alkarex** [comment, marked as answer]
Hello,
Which address are you typing in Readrops?

For that app, try to type just the root URL of your FreshRSS instance, like `https://freshrss.example.net` or `https://example.net/FreshRSS/` without the API path.

Did you check the automatic test of your API? `https://freshrss.example.net/api/`

**Comment by 360fun** [comment]
It works!!! 🥳 super super thank you! 🙏 
somebody should write this somewhere because it's not obvious...in most clients you must put the gReader url 🙄

**Reply by sunjam** [reply]
This is exactly correct with [Readrops](https://github.com/readrops/Readrops) Android app. Thank you!  I had to use the full greader.php url for the root server.

**Comment by aledeg** [comment]
@360fun Documentation PR welcomed!

**Comment by aledeg** [comment]
Or at least, could you describe the step by step process you've followed. Thank you

**Comment by 360fun** [comment]
After @Alkarex suggestion is very easy:

**ADD A FRESHRSS ACCOUNT ON READROPS**
Account name: _Whaterver You Like_ (default is "FreshRSS")
Account url: _root URL of your FreshRSS instance_ (ex. https://freshrss.example.net or https://example.net/FreshRSS/ without the API path)
Login: _username_
Password: _API password_ (not the account password!)

Hope this is enough! 🙏

**Comment by zolointo** [comment]
Thank you! I used part of this to solve my authentication issue, but wanted to make a documentation addition:

For this to work, you need to 1) set an API password for the account and 2) Allow API access

Quick instructions
**1. Set an API Password**
- login to FreshRSS on your browser
- navigate to your Profile page: Gear icon in upper-right -> **Profile**
- navigate to the bottom of this screen to the **API management** section
- type in your desired API password into the **API password** box
- click Submit

Once you do this, the API password will be recorded to your profile. **However**, the API password box will always be blank. This is what threw me, as I thought there was a FreshRSS error which was losing my password every time I clicked Submit. 

It is just a way to type in and submit a password. Just because it is blank doesn't mean there isn't an API password present.

2. Allow API Access
- login to FreshRSS on your browser
- navigate to the Authentication page: Gear icon in upper-right -> **Authentication**
- click the box next to **Allow API Access**
- click Submit
